### PR TITLE
Add snackbar and readme

### PR DESCRIPTION
**Pull request by Maianki**
## Added Snackbar and readme file
- [README.md](https://github.com/Maianki/ShuttleMotion-Video-Library/compare/add-snackbar-and-readme?expand=1#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5)
- [src/components/Card/VideoCard/VideoCard.jsx](https://github.com/Maianki/ShuttleMotion-Video-Library/compare/add-snackbar-and-readme?expand=1#diff-9445d61cedb1fcac13e59aff7c1549fec15b9f242551ade7b38996e9f1c46f54)
- [src/components/PlaylistModal/PlaylistModal.jsx](https://github.com/Maianki/ShuttleMotion-Video-Library/compare/add-snackbar-and-readme?expand=1#diff-9445d61cedb1fcac13e59aff7c1549fec15b9f242551ade7b38996e9f1c46f54)
- [src/components/Sanckbar/Snackbar.jsx](https://github.com/Maianki/ShuttleMotion-Video-Library/compare/add-snackbar-and-readme?expand=1#diff-9445d61cedb1fcac13e59aff7c1549fec15b9f242551ade7b38996e9f1c46f54)
- [src/context/auth-context.js](https://github.com/Maianki/ShuttleMotion-Video-Library/compare/add-snackbar-and-readme?expand=1#diff-9445d61cedb1fcac13e59aff7c1549fec15b9f242551ade7b38996e9f1c46f54)
- [src/context/playlists-context.js](https://github.com/Maianki/ShuttleMotion-Video-Library/compare/add-snackbar-and-readme?expand=1#diff-9445d61cedb1fcac13e59aff7c1549fec15b9f242551ade7b38996e9f1c46f54)
- [src/context/snackbar-context.js](https://github.com/Maianki/ShuttleMotion-Video-Library/compare/add-snackbar-and-readme?expand=1#diff-9445d61cedb1fcac13e59aff7c1549fec15b9f242551ade7b38996e9f1c46f54)
- [src/context/videos-operations-context.js](https://github.com/Maianki/ShuttleMotion-Video-Library/compare/add-snackbar-and-readme?expand=1#diff-9445d61cedb1fcac13e59aff7c1549fec15b9f242551ade7b38996e9f1c46f54)
- [src/hooks/useFilteredData.js](https://github.com/Maianki/ShuttleMotion-Video-Library/compare/add-snackbar-and-readme?expand=1#diff-9445d61cedb1fcac13e59aff7c1549fec15b9f242551ade7b38996e9f1c46f54)
- [src/pages/Playlists/Playlists.jsx](https://github.com/Maianki/ShuttleMotion-Video-Library/compare/add-snackbar-and-readme?expand=1#diff-9445d61cedb1fcac13e59aff7c1549fec15b9f242551ade7b38996e9f1c46f54)
- [src/pages/Signup/Signup.jsx](https://github.com/Maianki/ShuttleMotion-Video-Library/compare/add-snackbar-and-readme?expand=1#diff-9445d61cedb1fcac13e59aff7c1549fec15b9f242551ade7b38996e9f1c46f54)
- [src/pages/WatchVideo/WatchVideo.jsx](https://github.com/Maianki/ShuttleMotion-Video-Library/compare/add-snackbar-and-readme?expand=1#diff-9445d61cedb1fcac13e59aff7c1549fec15b9f242551ade7b38996e9f1c46f54)
- [src/reducers/snackbar-reducer.js](https://github.com/Maianki/ShuttleMotion-Video-Library/compare/add-snackbar-and-readme?expand=1#diff-9445d61cedb1fcac13e59aff7c1549fec15b9f242551ade7b38996e9f1c46f54)
### Live preview link
[Click here for live preview link](https://shuttle-motion-video-library-git-add-snackbar-an-97e12f-maianki.vercel.app/)

### CODE REVIEW

File: src/components/PlaylistModal/PlaylistModal.jsx [hunk 1]
```diff
@@ -2,14 +2,19 @@ import React, { useState } from "react";
 import styles from "./playlistmodal.module.css";
```
Comment (rahulyadav139) [inline src/components/PlaylistModal/PlaylistModal.jsx:2]: appreciation: great you are using CSS Modules.

File: src/pages/VideoListing/VideoListing.jsx [hunk 2]
```diff
@@ -31,19 +32,21 @@ export function VideoListing() {
...
-          ))}
+          {categories &&
+            categories.map(({ _id, categoryName }) => (
+              <Chips
+                key={_id}
+                categoryName={categoryName}
+                clickHandler={activeCategoryHandler}
+                activeCategory={activeCategory}
+              />
+            ))}
```
Comment (rahulyadav139) [inline src/pages/VideoListing/VideoListing.jsx:43]: Everything looks clean and readable. Great work.

**Review by rahulyadav139 — APPROVED** [review]

File: src/components/Card/VideoCard/VideoCard.jsx [hunk 3]
```diff
@@ -7,6 +7,7 @@ import { useLocation, Link } from "react-router-dom";
 import { getTrimVideoTitle } from "utils";
 import { useVideosOperations, usePlaylists } from "context";
 import { PlaylistModal } from "components";
+import { MdPlaylistPlay } from "react-icons/md";
```
Comment (ivinayakg) [inline src/components/Card/VideoCard/VideoCard.jsx:10]: nice using a third party library

File: src/components/PlaylistModal/PlaylistModal.jsx [hunk 4]
```diff
@@ -58,23 +67,24 @@ export function PlaylistModal({ btnModalHandler, video }) {
...
-                  )}
-                  onChange={(e) => playlistChangeHandler(e, _id)}
-                />
-                <label className='form-label-inline' htmlFor={title}>
-                  {title}
-                </label>
-              </div>
-            );
-          })}
+          {pathname !== "/playlist" &&
```
Comment (ivinayakg) [inline src/components/PlaylistModal/PlaylistModal.jsx:70]: take it to another component like playlist.jsx

File: src/components/Sanckbar/Snackbar.jsx [hunk 5]
```diff
@@ -0,0 +1,29 @@
+import React from "react";
```
Comment (ivinayakg) [inline src/components/Sanckbar/Snackbar.jsx:1]: no need to import React , its optional

**Review by ivinayakg — APPROVED** [review]
first thing , dont add files which are not related to the PR you are raising, and also don't add edited files in the PR either it causes confusion,
apart from that, the code is good structured, Good Job!

**Review by srishtigautam21 — COMMENTED** [review]
image
In the watch later page i think you forgot to change watch later to remove from watch later

**Review by srishtigautam21 — APPROVED** [review]
The code is well readable
Good to merge

File: src/components/Sanckbar/Snackbar.jsx (hunk 5 above)
Comment (Maianki) [reply src/components/Sanckbar/Snackbar.jsx:1]: Thanks for pointing that out.

**Comment by Maianki** [comment]
> image In the watch later page i think you forgot to change watch later to remove from watch later

Will change it thanks for letting me know that.

File: src/components/PlaylistModal/PlaylistModal.jsx (hunk 4 above)
Comment (Maianki) [reply src/components/PlaylistModal/PlaylistModal.jsx:70]: I need that in videocard.

**Comment by Maianki** [comment]
Thanks for reviewing @srishtigautam21 @ivinayakg @rahulyadav139

**Pull request merged**